// app/email-helper/index.js
var nodemailer = require('nodemailer'),
    email = 'pelmenforgsoap%40gmail.com',
    password = 'pelmengo12',
    transporter = nodemailer.createTransport('smtp://' + email + ':' + password + '@smtp.gmail.com'),
    fromAddress= '"Gsoap" <mail@gsoap.ru>',
    toAddress = 'mail@gsoap.ru';

module.exports = {
    sendEmail: function(subject, text, callback) {
        var mailOptions = {
            from: fromAddress,
            to: toAddress,
            subject: subject,
            text: text
        };

        transporter.sendMail(mailOptions, function(error, info) {
            if(callback) {
                callback(error, info);
            }
        });
    },
    sendReviewEmail: function(review, product, callback) {
        var subject = 'Добавлен отзыв на продукт ' + product.name,
            text = review.customerName + ' пишет: \n' +
                review.text + '\n' +
                'Email клиента: ' + review.customerEmail + '\n' +
                'Ссылка на продукт: ' + 'gsoap.ru/#/!/products/' + (product.id || product._id);
        this.sendEmail(subject, text, callback);
    },
    sendOrderEmail: function(order, callback) {
        var subject = 'Новый заказ: ' + order.orderIndex,
            text = 'Номер заказа: ' + order.orderIndex + '\n\n';
        text += 'Ссылка на заказ: http://gsoap.ru/admin#/orders/' + order.id + '/\n\n';
        text += 'Состав заказа:\n\n';
        order.products.forEach(function(product) {
            text += [product.name, 'емкость : ' + product.capacityInfo.capacity + 'г',
                    'цена : ' + product.capacityInfo.price + 'р', 'количество : ' + product.count].join(', ') + '\n';
        });
        text += '\nКонтактные данные:\n\n' +
                'Имя: ' + order.customerInfo.name + '\n' +
                'Email: ' + order.customerInfo.email + '\n' +
                'Телефон: ' + order.customerInfo.phone + '\n' +
                (order.customerInfo.promocode ? 'Номер купона на скидку: ' + order.customerInfo.promocode + '\n' : '') +
                'Способ доставки: ' + (order.customerInfo.deliveryType === 'self' ? 'Самовывоз, г. Тула' : 'Дотавка по РФ') + '\n' +
                'Адрес: ' + order.customerInfo.fullAddress;
        this.sendEmail(subject, text, callback);
    }
};
